File: test/samples/manualformat/manualformat.c
#include <stdio.h>
#include <stdlib.h>

/// \param a a first number
/// \param b a second number
/// \param c a third number
/// \param n This one on a new line
/// \param one_per_line an integer list, one per line
void manual_format(int a, int b, int c, int n, int* one_per_line) {
    /* TODO From the function perspective, this is just 4 integers */
}

int main() {
    int a;
    scanf("%d", &a);
    int b;
    scanf("%d", &b);
    int c;
    scanf("%d", &c);
    int n;
    scanf("%d", &n);
    int* one_per_line = (int*)malloc(3 * sizeof(int));
    for (int i = 0; i < 3; ++i)
        scanf("%d", &one_per_line[i]);
    manual_format(a, b, c, n, one_per_line);

    return 0;
}

File: test/samples/manualformat/manualformat.pas
program ManualFormat;

type
    T_OnePerLine = array of longint;

{ @param A a first number }
{ @param B a second number }
{ @param C a third number }
{ @param N This one on a new line }
{ @param OnePerLine an integer list, one per line }
procedure ManualFormat(A: longint; B: longint; C: longint; N: longint; const OnePerLine: T_OnePerLine);
begin
    {* TODO From the function perspective, this is just 4 integers *}
end;

var
    A: longint; { a first number }
    B: longint; { a second number }
    C: longint; { a third number }
    N: longint; { This one on a new line }
    OnePerLine: T_OnePerLine; { an integer list, one per line }
    i: longint;
begin
    readln(A, B, C);
    readln(N);
    setLength(OnePerLine, 3);
    for i := 0 to 3 - 1 do
    begin
        readln(OnePerLine[i]);
    end;
    ManualFormat(A, B, C, N, OnePerLine);
end.
